spec/specs/engine_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'spec_helper'
+require 'metricks/engine'
+
+describe Metricks::Engine do
+
+  let(:mock_app) do
+    Class.new(Rails::Application) do
+      config.eager_load = false
+    end
+  end
+
+  before do
+    allow(Metricks::Lock).to receive(:validate!).and_call_original
+  end
+
+  it 'allows with_lock to be configured' do
+    success = false
+
+    allow(mock_app.config.metricks).to receive(:with_lock)
+      .and_return(->(result, opts, block) { block.call(result, opts) })
+
+    expect {
+      mock_app.initialize!
+    }.not_to raise_error
+
+    Metricks::Lock.with_lock(true, {}) do |result|
+      success = result
+    end
+
+    expect(success).to be(true)
+
+    expect(Metricks::Lock).to have_received(:validate!)
+  end
```

Comment:
In an ideal world we might test various different rails app configurations. I did try to get that working, but Rails basically doesn't expect multiple apps to be declared and they were leaking into different tests. In the end I went with just verifying the override works and that we call the expected validation.